src/block/block_ckpt.c

```diff
@@ -671,7 +721,7 @@ __ckpt_process(WT_SESSION_IMPL *session, WT_BLOCK *block, WT_CKPT *ckptbase)
      * Record the checkpoint's allocated blocks. Do so before skipping any processing and before
      * possibly merging in blocks from any previous checkpoint.
```

Comment:
I thought about this for a bit. I am pretty sure it is correct for discarded blocks too.

At this point we have the alloc and discard lists for the new checkpoint -- i.e., the allocate changes relative to the previous checkpoint. And that is the only information we want to update in the incremental block map(s). I.e., we don't want additional alloc/discard information from deleted checkpoints since that should already be in the bitmaps.

No action required here. Just thinking out loud so @sueloverso and other reviewers can jump in if I'm misunderstanding.